frontend/public/components/storage-class-form.tsx

```diff
@@ -857,13 +845,32 @@ class StorageClassFormWithTranslation extends React.Component<
       const isCheckbox = parameter.type === 'checkbox';
       const selectedKey = ['newStorageClass', 'parameters', key, 'value'];
 
-      if (parameter.visible && !parameter.visible(this.state.newStorageClass.parameters)) {
+      if (isExecutableCodeRef(parameter?.visible)) {
+        executeReferencedFunction(parameter?.visible, this.state.newStorageClass.parameters).then(
+          (isVisible) => {
+            if (isVisible) {
+              return null;
+            }
+          },
+        );
+      }
+      if (
+        (typeof parameter?.visible === 'function' &&
+          !parameter.visible(this.state.newStorageClass.parameters)) ||
+        parameter?.visible === false
+      ) {
         return null;
       }
 
       if (parameter.Component) {
         const { Component } = parameter;
-        return (
+        return isExecutableCodeRef(Component) ? (
+          <AsyncComponent
+            loader={Component}
```

Comment:
Component can only be a coderef, therefore the else condition can be removed.

Reply:
We can add components directly to this file as well. In that case it wouldn't be code ref.

Reply:
I wasn't aware. Why not register all through extensions?

Reply:
Well plugins could contribute storage classes as well, but there are default storage classes supported by OCP which can have these components embedded in the file itself.

Reply:
@bipuladh but why wouldn't core console contribute these defaults through the same mechanism?

Reply:
https://github.com/openshift/console/blob/master/frontend/public/components/storage-class-form.tsx#L149
So one way we could enforce this is by making sure that entries here in the future are lazy imports, else we need the above block. We are adding other storage class provisioners directly from the same file and not frome console-app.

Reply:
@bipuladh you haven't answered my question still.
Why add the defaults here instead of contributed through the same extensions?